syntax = "proto3";
package server_services;

service ImporterService {
    rpc UploadCSV (FileUploadRequest) returns (FileUploadResponse);
    rpc ConvertToXML (CSVDataRequest) returns (XMLFileResponse);
}

message FileUploadRequest {
    bytes file = 1; // DTD file as bytes
    string file_mime = 2;
    string file_name = 3;
}

message FileUploadResponse {
    bool success = 1; 
}

message CSVDataRequest {
    string file_name = 1;
    string file_mime = 2; 
}

message XMLFileResponse {
    bool success = 1; 
}

// Service definition
service SendFileService {
 rpc SendFile (SendFileRequestBody) returns (SendFileResponseBody);
}
// Request message
message SendFileRequestBody {
 bytes file = 1; // DTD file as bytes
 string file_mime = 2;
 string file_name = 3;
}

// Response message
message SendFileResponseBody {
 bool success = 1;
}